Snowplow/Internal/Session/SPSession.m

```diff
@@ -140,7 +142,12 @@ - (NSDictionary *) getSessionDictWithEventId:(NSString *)eventId {
             }
             self.lastSessionCheck = [SPUtilities getTimestamp];
         }
-        return _state.sessionContext;
+        
+        _eventIndex += 1;
+        
+        NSMutableDictionary *context = _state.sessionContext;
+        [context setObject:[NSNumber numberWithInteger:_eventIndex] forKey:kSPSessionEventIndex];
```

Comment:
👍 👍 
I think another possible approach could be creating a new SessionState each time. It seems expensive but at the end it wouldn't change much as sessionContext is cloned each time. In that case we should consider also the implication it has with the session callback. Anyway, it's something we can reconsider in case we adopt the FSM concept. Just a thought for a future discussion.

Reply:
That's a good point! I think it would make sense to have an immutable SessionState that we clone each time. In this case, I wanted to make this change outside of the SessionState because the eventIndex property is not persisted in database along with the rest of the SessionState but only kept in-memory in SPSession, so I wanted to add it separately.

Reply:
> I wanted to make this change outside of the SessionState because the eventIndex property is not persisted in database along with the rest of the SessionState

👍  good point